### PR TITLE
The `wasm_bindgen` support of `fuel-asm` and `fuel-types`

### PR DESCRIPTION
**Pull request by xgreenx**
Added `wasm_bindgen` for `fuel-types` and `fuel-asm`. The `wasm_bindgen` requires C-style enums. So I removed all constant variables and merged the enum definition with the `TryFrom` or `From` implementation to use the values directly.

 The `wasm_bindgen` doesn't support non C-style enums, so I created a wrapper around `Instruction` - `typescript::Instruction`.

 The `wasm_bindgen` doesn't work with constant functions and generics, so I created non-constant and non-generic analogs for methods.

 The `wasm_bindgen` doesn't support constants, so I created non-constant methods.

Replaced the `Self` with actual type's name because it is required by `wasm_bindgen`.

I removed the usage of `fuel-types` from `fuel-asm` to generate less code for `fuels-ts`.

### CODE REVIEW

File: fuel-types/src/array_types.rs [hunk 1]
```diff
@@ -130,6 +132,29 @@ macro_rules! key_methods {
...
+            /// Bytes constructor.
+            pub fn from_bytes(bytes: &[u8]) -> Self {
+                Self(bytes.try_into().expect(
+                    format!("The size of the arrays it not {} size", $s).as_str(),
+                ))
+            }
+
+            /// Zeroes bytes constructor.
+            #[wasm_bindgen(js_name = zeroed)]
+            pub fn zeroed_typescritp() -> $i {
```
Comment (Dentosal) [inline fuel-types/src/array_types.rs:147]: `typescritp` -> `typescript`

File: fuel-types/src/array_types.rs [hunk 2]
```diff
@@ -130,6 +132,29 @@ macro_rules! key_methods {
             }
         }
 
+        #[cfg(feature = "typescript")]
+        #[wasm_bindgen::prelude::wasm_bindgen]
+        impl $i {
+            /// Bytes constructor.
```
Comment (Dentosal) [inline fuel-types/src/array_types.rs:138]: Documentation of the panic condition missing

**Review by Dentosal — CHANGES_REQUESTED** [review]
Only minor issues, but lets fix those before merging

**Comment by arboleya** [comment]
@xgreenx How can I build these two locally?

I'd assume it is done via `wasm-pack` _(?)_, but I couldn't find any single-liner to do it with the CI scripts.

**Comment by xgreenx** [comment]
One time install:
```shell
cargo install wasm-bindgen-cli
``

And to build the rust to WASM and WASM to js/ts:
```shell 
cargo build -p fuel-types --target wasm32-unknown-unknown --features typescript --release
cargo build -p fuel-asm --target wasm32-unknown-unknown --features typescript --release
wasm-bindgen --target nodejs --typescript ./target/wasm32-unknown-unknown/release/fuel_types.wasm --out-dir ./typescript_types
wasm-bindgen --target nodejs --typescript ./target/wasm32-unknown-unknown/release/fuel_asm.wasm --out-dir ./typescript_asm
```

**Review by Voxelot — DISMISSED** [review]

**Review by Dentosal — APPROVED** [review]

**Review by arboleya — DISMISSED** [review]

**Review by arboleya — APPROVED** [review]

**Pull request merged**